### PR TITLE
Added a SegDataset class for the segmentation task

### PR DESCRIPTION
**Pull request by msminhas93**
Added a class for handling segmentation data. It gives you access to image,mask pairs. I've defined 4 transforms for the same: (1) Resize (2) ToTensor (3) Normalize (4) Lambda. Additionally, there are two helper functions for creating dataloaders either from a single dataset folder or from separate 'Train' and 'Test' folders. @fmassa your feedback would be greatly appreciated. If there are any suggestion I'll be happy to discuss those and contribute to the community. #424 #1330 #1315

### CODE REVIEW

**Comment by SebastienEske** [comment]
Hello, I just made this [pull request](https://github.com/pytorch/vision/pull/1345) which is probably more generic and uses torchvision transforms
